edu.wpi.first.wpilib.plugins.cpp/src/main/java/edu/wpi/first/wpilib/plugins/cpp/WPILibCPPPlugin.java

```diff
@@ -81,8 +99,101 @@ public void earlyStartup() {
     	WPILibCore.getDefault().saveGlobalProperties(props);
+		updateProjects();
+	}
+	
+	public void updateProjects() {
+		WPILibCPPPlugin.logInfo("Updating projects");
+		
+		// Get the root of the workspace
+	    IWorkspace workspace = ResourcesPlugin.getWorkspace();
+	    IWorkspaceRoot root = workspace.getRoot();
+	    // Get all projects in the workspace
+	    IProject[] projects = root.getProjects();
+	    // Loop over all projects
+	    for (IProject project : projects) {
+			  try {
+				  if(project.hasNature("edu.wpi.first.wpilib.plugins.core.nature.FRCProjectNature") && project.hasNature("org.eclipse.cdt.core.ccnature")){
+					updateVariables(project);
+				  }
+			  } catch (CoreException e) {
+				WPILibCPPPlugin.logError("Error updating projects.", e);
+			  }
+	    }
+	}
+	
+	public void updateVariables(IProject project) {
+		final IManagedBuildInfo buildInfo = ManagedBuildManager.getBuildInfo(project, true);
+		final IConfiguration config = buildInfo.getDefaultConfiguration();
+		final ITool[] tools = config.getRootFolderInfo().getTools();
+		
+		for (final ITool tool : tools) {
+			if (tool.getId().contains(".cpp.linker.")) {
+				IOption option = tool.getOptionBySuperClassId("gnu.cpp.link.option.libs");
+				if(option != null) {
+					try {
+						String[] libs = option.getLibraries();
+						List<String> libList = new ArrayList<String>();
+						
+						File dir = new File(WPILibCore.getDefault().getWPILibBaseDir() + File.separator + "user" + File.separator + "cpp" + File.separator + "lib");
+						File[] filesList = dir.listFiles();
+						for (File file : filesList) {
+							if (file.isFile()) {
+								String[] splitFileName = file.getName().substring(3).split("[.]");
+								if(splitFileName[splitFileName.length-1].equals("so") || splitFileName[splitFileName.length-1].equals("a"))
+								{
+									String libName = file.getName().substring(3, file.getName().length()-splitFileName[splitFileName.length-1].length()-1);
+									if(!libList.contains(libName))
+									{
+										libList.add(libName);
+										WPILibCPPPlugin.logInfo("Adding library to cpp link: " + file.getName());
+									}
+								}
+							}
+						}
+						libList.add("wpi");
+						option.setValue(libList.toArray(new String[libList.size()]));
+					} catch (final BuildException e) {
+						WPILibCPPPlugin.logError("Error retrieving library information from tool option", e);
+					}
+				}
+				
+				try {
+					option = tool.getOptionBySuperClassId("gnu.cpp.link.option.paths");
+					String[] paths = option.getBasicStringListValue();
+					List<String> libPathsList = new ArrayList<String>(Arrays.asList(paths));
+					if(!libPathsList.contains(USER_LIBS_PATH))
+					{
+						libPathsList.add(USER_LIBS_PATH);
+						option.setValue(libPathsList.toArray(new String[libPathsList.size()]));
+					}
+				} catch (final BuildException e) {
+					WPILibCPPPlugin.logError("Error checking library paths", e);
+				}
+				/*
```

Comment:
It's the code to set linker flags per the rpath TODO at the top. I'll add a comment to indicate that